hometask-2/src/data-access/user-repository.ts

```diff
@@ -0,0 +1,80 @@
+import { Op } from 'sequelize';
+import { v4 as uuidv4 } from 'uuid';
+
+import { User } from '../types';
+import { getUserModel } from './models/user';
+import { normalizeUser, normalizeUserRecord } from './normalizers/user';
+
+export async function getUserById(id: string): Promise<User | null> {
+    const user = await getUserModel().findOne({
+        where: { id, is_deleted: false },
+        raw: true
+    });
+
+    return normalizeUser(user);
+}
+
+export async function addUser(user: Omit<User, 'id | isDeleted'>): Promise<string> {
+    const generatedId = uuidv4();
+    const newUser = { ...user, id: generatedId, isDeleted: false };
+
+    const response = (await getUserModel().create(normalizeUserRecord(newUser))).get({ plain: true });
+
+    return response?.id;
+}
+
+export async function updateUser(id: string, user: Partial<User>): Promise<User | null> {
+    const response = await getUserModel().update(user, {
+        fields: [
+            'login', 'password', 'age'
+        ],
+        where: {
+            id,
+            is_deleted: false
+        },
+        returning: true
+    });
+
+    const updatedUser = response[1][0];
+
+    return normalizeUser(updatedUser);
+}
+
+export async function deleteUser(id: string): Promise<string> {
+    const response = await getUserModel().update({ is_deleted: true }, {
+        fields: [
+            'is_deleted'
+        ],
+        where: {
+            id,
+            is_deleted: false
+        },
+        returning: true
+    });
+
+    const softDeletedUser = response?.[1]?.[0];
+
+    return softDeletedUser?.id;
+}
+
+export async function getAllUsers(limit?: number) {
+    const users = await getUserModel().findAll({
+        where: { is_deleted: false },
+        limit: limit || undefined,
```

Comment:
I know that it does not matter much for the homework, but a good rule of thumb is always limit the queries, otherwise you will be open to malicious DoS attacks, or even the normal use of the application might degrade the performance as the dataset grows

Reply:
Added default limit value